src/main/java/seedu/address/model/person/Attendance.java

```diff
@@ -1,58 +1,80 @@
 package seedu.address.model.person;
 
 /**
- * The `Attendance` class represents a student's attendance and participation record for a 12-week period.
+ * The `Attendance` class represents a student's attendance, attendance status
+ * and participation record for a 12-week period.
  * It provides methods for managing and querying attendance information.
  */
 public class Attendance {
     public static final String TUTORIAL_ERROR_MSG = "Tutorial number is out of range, should be integer between 1-12";
+    public static final String STATUS_ERROR_MSG = "Status is invalid! It should either be P, VR or A.";
+    public static final String ORIGINAL_ATD = "U,U,U,U,U,U,U,U,U,U,U,U";
     public static final String PARTICIPATION_ERROR_MSG = "Please input a small non-negative number.";
-    public static final String ORIGINAL_ATD = "0,0,0,0,0,0,0,0,0,0,0,0";
     public static final String ORIGINAL_PART = "0,0,0,0,0,0,0,0,0,0,0,0";
-    private int totalTut;
-    private final boolean[] attendanceList = new boolean[12];
-    private final int[] participationList = new int[12];
+    private static int totalTut = 12;
+    private final int[] participationList = new int[totalTut];
+    private final AttendanceStatus[] attendanceList = new AttendanceStatus[totalTut];
+
+    private enum AttendanceStatus {
+        UNMARKED,
+        PRESENT,
+        ABSENT,
+        VR;
+
+        @Override
+        public String toString() {
+            return this == PRESENT ? "P" : this == ABSENT ? "A" : this == VR ? "VR" : "U";
```

Comment:
This is unsound style, which is unreadable and unmaintainable. Would it be better to use `if-else` statements or even better, a `case` `switch` statement?